### PR TITLE
Prabhav

### PR DESCRIPTION
**Pull request by probro27**
# Strategy Onboarding

## TODO List

Onboarding tasks:

- [x] Task 1: Implement `fibonacci_term()` in `fibonacci.py`
  - All tests in `test_fibonacci.py` should pass, run `pytest` command to check
- [x] Task 2: Implement `find_min_index()` and `reverse_str_arr()`
  - Write tests that cover the logic that you wrote in `test_array.py`
- [x] Task 3: Process and visualize sample data in `matplot.py`
  - Read data in CSV using `pandas` or `csv`, handle the missing values, plot using `matplotlib`
  - Include a screenshot of your graph in the PR

Some extra items:

- [x] Completed [strategy welcome survey](https://forms.gle/xnYBumyAn2mA2Ao68)
- [x] Added `@e-wai` as a reviewer (top of right column in the "Open a pull request" page)

### CODE REVIEW

File: onboarding/tests/test_array.py [hunk 1]
```diff
@@ -1 +1,12 @@
...
+from ..array import find_min_index, reverse_str_arr
+
+def test_find_min_index():
+    assert(find_min_index([1,2,3,4,5]) == 0)
+    assert(find_min_index([0,0,0,0,0,0]) == 0)
+    assert((find_min_index([543,234,23,54,23]) == 2))
+
+def test_reverse_str_arr():
+    assert(reverse_str_arr('HELLO') == ['O', 'L', 'L', 'E', 'H'])
+    assert(reverse_str_arr('AAAAAA') == ['A', 'A', 'A', 'A', 'A', 'A'])
```
Comment (e-wai) [inline onboarding/tests/test_array.py:12]: nitpick: newline at end of file please!

File: pull_request_template.md [hunk 2]
```diff
@@ -22,3 +22,4 @@ Some extra items:
 
 Feel free to put any questions, comments, or concerns here. If you have anything you want reviewers to double-check, also good to flag it here!
 -->
+
```
Comment (e-wai) [inline pull_request_template.md:25]: Unnecessary change?

File: onboarding/fibonacci.py [hunk 3]
```diff
@@ -15,4 +15,19 @@ def fibonacci_term(n):
...
+        c = a + b
+        a = b
+        b = c
+      return c
+    else:
+      print("Please enter a value greater than 0")
+      raise TypeError
+
+
+# print(fibonacci_term(-1))
```
Comment (e-wai) [inline onboarding/fibonacci.py:33]: Can you remove these lines?

File: onboarding/matplot.py [hunk 4]
```diff
@@ -5,3 +5,23 @@
 - Be careful of data points that don't exist
 - Will require reading from CSV, can experiment with pandas
 '''
+
+import pandas as pd
+import matplotlib.pyplot as plt
```
Comment (e-wai) [inline onboarding/matplot.py:10]: You need to add the package imports to `requirements.txt` so that it gets installed with the repo on a `pip install -r requirements.txt`

File: onboarding/array.py [hunk 5]
```diff
@@ -24,4 +31,8 @@ def reverse_str_arr(string):
     '''
 
     # insert code here
-    pass
\ No newline at end of file
+    letters = []
+    for i in string:
+        letters.insert(0, i)
+    
+    return letters
```
Comment (e-wai) [inline onboarding/array.py:38]: Newline at end of file please!

File: onboarding/array.py [hunk 6]
```diff
@@ -9,7 +9,14 @@ def find_min_index(arr):
     '''
 
     # insert code here
-    pass
+    min = arr[0]
+    minpos = 0
```
Comment (e-wai) [inline onboarding/array.py:13]: Super super nitpicky, but general convention is snakecase so for legibility I would always encourage `min_pos` 😀

File: onboarding/array.py [hunk 7]
```diff
@@ -9,7 +9,14 @@ def find_min_index(arr):
     '''
 
     # insert code here
-    pass
+    min = arr[0]
```
Comment (e-wai) [inline onboarding/array.py:12]: What happens if array is empty?

File: onboarding/tests/test_array.py [hunk 8]
```diff
@@ -1 +1,12 @@
 # write some tests here!
+import pytest
+from ..array import find_min_index, reverse_str_arr
+
+def test_find_min_index():
+    assert(find_min_index([1,2,3,4,5]) == 0)
+    assert(find_min_index([0,0,0,0,0,0]) == 0)
+    assert((find_min_index([543,234,23,54,23]) == 2))
```
Comment (e-wai) [inline onboarding/tests/test_array.py:8]: Can you try out a couple negative values, different array sizes (eg. size 0)?

**Review by e-wai — CHANGES_REQUESTED** [review]
It's looking pretty good! Some quick suggestions here and there

File: pull_request_template.md (hunk 2 above)
Comment (probro27) [reply pull_request_template.md]: yeah it was by mistake.

File: onboarding/array.py (hunk 7 above)
Comment (probro27) [reply onboarding/array.py:12]: Since there was nothing given, i considered that the array is non-empty. If it is not then could u suggest what I should return? Will None be fine?

Comment (e-wai) [reply onboarding/array.py:15]: That works 👍

File: onboarding/tests/test_array.py (hunk 8 above)
Comment (e-wai) [reply onboarding/tests/test_array.py:8]: ^ feel free to resolve conversations when they're addressed

**Review by e-wai — APPROVED** [review]
